code/client/munkilib/fetch.py

```diff
@@ -305,10 +306,18 @@ def get_url(url, destinationpath,
 
     if str(connection.status).startswith('2') and temp_download_exists:
         os.rename(tempdownloadpath, destinationpath)
+        if postware:
+            display.display_debug2('Processing options through postware')
+            # postware module must have process_post_fetch function
+            postware.process_post_fetch(url, destinationpath)
         return connection.headers
     elif connection.status == 304:
         # unchanged on server
         display.display_debug1('Item is unchanged on the server.')
+        if postware:
+            display.display_debug2('Processing options through postware')
+            # postware module must have process_post_fetch function
+            postware.process_post_fetch(url, destinationpath)
```

Comment:
If the item is unchanged, the signature was checked during the initial download (unless you've just switched to checking signatures). Is there a good reason to check the signature again?

Reply:
I think the idea was that if the signature changed but the download itself hadn't, that we should re-run the postware (in the case of the signature-checking). Due to the fact we place it in a temp file the signature is always re-downloaded, however. Perhaps another postware function should be called (or a different argument/option supplied to the function to indicate the separate calls)? Or perhaps we should save the signaures along-side the cats/manifests? This way we could not download them over and over again.

I don't want the "postware" to be exclusively for signature checking - hopefully it's generic enough for other use cases, too.